### PR TITLE
Add ability for mapping floats to reversed ranges

### PR DESCRIPTION
**Pull request by ZevEisenberg**
@heyltsjay check it out. Not sure if it's worth it, but this was fun to put together.

### CODE REVIEW

**Comment by heyltsjay** [comment]
@ZevEisenberg I think my vote may just be for a defaulted boolean `inverted`. I bet when people consume this, they will figure out their ranges through math / prints / debugging, and will plug them in and go. They'll then see that the relationship is backwards and will just want to flip something. I think it's more important to indicate that the relationship is inverted than implying that the directionality of either argument is important. (both descending == both ascending, either one descending == same result)

```titleView.offset = scrollView.contentOffset.y.scaled(from: 100...150, to: 0...30, inverted: true, clamped: true)```

**Comment by ZevEisenberg** [comment]
@heyltsjay great point about the direction of both arguments. With my API, there are four permutations of direction (↑↑, ↑↓, ↓↑, ↓↓), but only two distinct outcomes (normal, reversed). We should almost definitely switch to your version.

**Comment by jvisenti** [comment]
@ZevEisenberg If the map function were to take a `reversed` `Bool`, then there would be no need for a `ReversedClosedRange` right?

**Comment by ZevEisenberg** [comment]
@jvisenti right. It significantly reduces the API surface. Then, just reverse the upper and lower of one of either the "from" or "to" ranges. I'll do that and push it to this branch.

**Comment by ZevEisenberg** [comment]
Updated. I also updated the parameter name from `clamp` to `clamped` to match the `clamped(to:)` API that it calls down to, and updated the doc comment. How's this?

```swift
/// Re-maps a number from one range to another.
///
/// - Parameters:
///   - source: The range to interpret the number as being a part of.
///   - destination: The range to map the number to.
///   - clamped: Whether the result should be clamped to the `to` range. Defaults to `false`.
///   - reversed: whether the output mapping should be revserd, such that
///               as the intput increases, the output decreases. Defaults to `false`.
/// - Returns: The input number, scaled from the `from` range to the `to` range.
```

**Comment by jvisenti** [comment]
Cleaner, but one downside is that you can't "reverse" the source range, only the destination. I don't think that matters much because you could just switch the params

**Comment by heyltsjay** [comment]
@jvisenti per the discussion the above, it doesn't matter which range is switched. there are only two possible mapping outcomes.

File: Pod/Classes/Math/FloatingPoint+Scale.swift [hunk 1]
```diff
@@ -13,18 +13,21 @@ public extension FloatingPoint {
     /// - Parameters:
     ///   - source: The range to interpret the number as being a part of.
     ///   - destination: The range to map the number to.
-    ///   - clamp: Whether the result should be clamped to the `to` range.
+    ///   - clamped: Whether the result should be clamped to the `to` range. Defaults to `false`.
+    ///   - reversed: whether the output mapping should be revserd, such that
+    ///               as the intput increases, the output decreases. Defaults to `false`.
```
Comment (heyltsjay) [inline Pod/Classes/Math/FloatingPoint+Scale.swift]: typo: `intput`

**Pull request merged**